### PR TITLE
Consent State Group & Member Associations and Bindings

### PR DESCRIPTION
**Pull request by nplasterer**
Part of https://github.com/xmtp/libxmtp/issues/801
This adds the bindings for inserting consent as well as fetching it on members and groups.

- [x] Bindings for consent state on group
- [x] Bindings for consent state on member
- [x] Binding for adding consent state to a group
- [x] Binding for adding any consent (to help with bulk import/migration)
- [x] Binding for fetching consent for an address, id, or group id (to help with V2/V3 migration)

### CODE REVIEW

**Comment by nplasterer** [comment]
I've got a lot of tests to write but would love some eyes on the approach I took her for consent.
Mostly trying to make it simple enough for getting the group or member consent quickly while also adding the legacy way to set and get to help with the dual sending period. Open to thoughts.

File: bindings_ffi/src/mls.rs [hunk 1]
```diff
@@ -1333,6 +1399,40 @@ impl FfiGroup {
...
+    }
+
+    pub async fn update_consent_state(&self, state: FfiConsentState) -> Result<(), GenericError> {
+        let group = MlsGroup::new(
+            self.inner_client.context().clone(),
+            self.group_id.clone(),
+            self.created_at_ns,
+        );
+
+        let consent_state = match state {
```
Comment (insipx) [inline bindings_ffi/src/mls.rs:1425]: Can make this into a separate implementation,

```

impl From<FfiConsentState> for ConsentState {
fn from(state: FfiConsentState) -> ConsentState {
        FfiConsentState::Unknown => ConsentState::Unknown,
            FfiConsentState::Allowed => ConsentState::Allowed,
            FfiConsentState::Denied => ConsentState::Denied,
}
}
```

Then wherever you're doing this match, can do `ConsentState::from(FfiConsentState)` or use the `into()` method, `let state: ConsentState = state.into()


doing this for the other enums as well should clean the code up a bit

File: xmtp_mls/src/groups/mod.rs [hunk 2]
```diff
@@ -942,6 +947,23 @@ impl MlsGroup {
             })
     }
 
+    /// Find the `consent_state` of the group
+    pub fn consent_state(&self) -> Result<ConsentState, GroupError> {
+        let conn = self.context.store.conn()?;
+        conn.find_group(self.group_id.clone())
+            .map_err(GroupError::from)
```
Comment (insipx) [inline xmtp_mls/src/groups/mod.rs:954]:
```suggestion
            .map_err(GroupError::from)?
```



Then you shouldn't need the `and_then`

**Review by insipx — COMMENTED** [review]
Looking good!

**Review by insipx — APPROVED** [review]

**Review by neekolas — APPROVED** [review]

**Pull request merged**